apps/ledger-live-desktop/src/renderer/hooks/usePortfolioCards.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import ActionCard from "~/renderer/components/ContentCards/ActionCard";
+import { portfolioContentCardSelector } from "~/renderer/reducers/dynamicContent";
+import * as braze from "@braze/web-sdk";
+import { setPortfolioCards } from "~/renderer/actions/dynamicContent";
+
+const usePortfolioCards = () => {
+  const dispatch = useDispatch();
+  const [cachedContentCards, setCachedContentCards] = useState<braze.Card[]>([]);
+  const portfolioCards = useSelector(portfolioContentCardSelector);
+
+  useEffect(() => {
+    const cards = braze.getCachedContentCards().cards;
+    setCachedContentCards(cards);
+  }, []);
+
+  const onImpression = (cardId: string) => {
+    const currentCard = cachedContentCards.find(card => card.id === cardId);
+    currentCard && braze.logContentCardImpressions([currentCard]);
+  };
+
+  const onDismiss = (cardId: string) => {
+    const currentCard = cachedContentCards.find(card => card.id === cardId);
+
+    if (currentCard) {
+      braze.logCardDismissal(currentCard);
+      setCachedContentCards(cachedContentCards.filter(n => n.id !== currentCard.id));
+      dispatch(setPortfolioCards(portfolioCards.filter(n => n.id !== currentCard.id)));
+    }
+  };
+
+  const onClick = (cardId: string) => {
+    const currentCard = cachedContentCards.find(card => card.id === cardId);
+
+    if (currentCard) {
+      // For some reason braze won't log the click event if the card url is empty
+      // Setting it as the card id just to have a dummy non empty value
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
We need to ban `// @ts-ignore` OR `any` from our TS usage with new features.
Lately, we had an issue on production with a TS error that we ignored. We need to find why` For some reason braze won't log the click` or extend the type.

Reply:
I 100% agree with you, there are a lot of "hidden" issues. This part of code isn't new, it's re-used code from the legacy piece of code that was handling braze before on LLD and left it as is.
I just dug a little and after few tests, all of the cards I receive from Braze have their `url` field set. So I'm just removing this part of code but again, I cannot fully know what will be the consequences of this code removal, hence the reason why I kept it in the first place.

FYI, on top that, the SDK we are using is almost 1 year old.